### ISSUE TITLE
boxed_local not detailed enough?

### ISSUE THREAD

**Issue by Keats** [Manishearth/rust-clippy#606]
Crossposting from reddit: https://www.reddit.com/r/rust/comments/42gh9h/hey_new_rust_users_got_an_easy_question_ask_it/czgy3qb

In short, running clippy on https://github.com/Keats/dbmigrate yields the following warning:

```
src/cmd.rs:46:15: 46:21 warning: local variable doesn't need to be boxed here, #[warn(boxed_local)] on by default
src/cmd.rs:46 pub fn status(driver: Box<Driver>, migration_files: &Migrations) {
```

with `Driver` here being a Trait. Maybe it should explain what are the alternatives as I thought Box<> was the only way to pass trait objects around ? I'm not sure what's the fix to do there

**Comment by jacobmorzinski** [comment]
What a coincidence - I'm working through [Too Many Linked Lists](http://cglab.ca/~abeinges/blah/too-many-lists/book/) and have the same confusion.  When I reach [Chapter 3.2 "Making it all Generic"](http://cglab.ca/~abeinges/blah/too-many-lists/book/second-generic.html) and follow the guide for a `pop` fn, clippy warns that "node" doesn't need to be boxed:

    src\second.rs:28:31: 28:35 warning: local variable doesn't need to be boxed here, #[warn(boxed_local)] on by default
    src\second.rs:28         self.head.take().map(|node| {
                                                   ^~~~
    src\second.rs:28:31: 28:35 help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#boxed_local

I'm confused because I don't know how `|node|` is boxed or how to unbox it.

**Comment by Manishearth** [comment]
Ah, we should be ignoring closure arguments, as well as boxed traits.

**Comment by llogiq** [comment]
I'd take it. Unless someone else with more time than me wants to step in.

**Comment by sanxiyn** [comment]
Fixed closure arguments part in #608.

**Comment by llogiq** [comment]
@Keats @jacobmorzinski Could you please re-test with the latest git master? I'd like to be sure my tests really capture your scenario.

**Comment by jacobmorzinski** [comment]
@llogiq Thank you for your help with this.  I'll try to test, but I'm not experienced with Rust compiler plugins.  Is there a guide, somewhere online, that teaches me how to compile and test a compiler plugin?  Are [this one](https://www.gulshansingh.com/posts/how-to-write-a-rust-syntax-extension/) or [that one](http://stackoverflow.com/questions/30418297/how-do-i-properly-register-a-rust-compiler-plugin) good places for me to start?

**Comment by llogiq** [comment]
Oh, I just meant could you please re-run clippy on your code to see if it still warns on boxed traits?

Though I have added tests that should make sure those no longer match, it would confirm my fix also works for code in the wild.

**Comment by jacobmorzinski** [comment]
If I change nothing in my other project's Cargo.toml, and I re-run `cargo build --features clippy`, it uses clippy from crates.io (v0.0.37), which still warns on boxed_local.

I can clone the git master of rust-clippy and can `cargo build` it, but I don't know what to do with it after I've built it.  I guess there might be a way for my other project's Cargo.toml to use the new clippy that I compiled and built, but I do not know how to tell one cargo project to use dependencies from a sibling cargo project.

After an hour of wrestling with this, I think I've solved it: what I am actually looking for is [the Overriding Dependencies section](http://doc.crates.io/guide.html#overriding-dependencies) of the Cargo Guide.  If I create a .cargo/config file, I can tell one project to depend on a sibling project.

So: yes!  The new test from the latest git master is good, and no longer warns about my closure argument.  Looks good.
